[BugFix] transaction stream load cancel bug

xiaoyong-z opened the pull request:
After we begin a transaction stream load and load, if we don't abort or commit for a long time,
transaction stream load will timeout, however, BE itself cannot cancel stream load.
This is because timeout_second hasn't been set if we don't explicitly specify timeout in stream load.

Signed-off-by: xyz <a997647204@gmail.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #https://github.com/StarRocks/StarRocksTest/issues/1629

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3
  - [ ] 2.2

trueeyu commented on be/src/runtime/fragment_mgr.cpp:
```diff
@@ -88,7 +89,7 @@ class RuntimeProfile;
 public:
     FragmentExecState(const TUniqueId& query_id, const TUniqueId& instance_id, int backend_num, ExecEnv* exec_env,
-                      const TNetworkAddress& coord_hostport);
+                      const TNetworkAddress& coord_hostport, const std::string stream_load_label = "");
```
`const std::string&` or `std::string`, otherwise the std::string will copy twice ?

trueeyu commented on gensrc/thrift/InternalService.thrift:
```diff
@@ -350,6 +350,8 @@ struct TExecPlanFragmentParams {
   // Sharing data between drivers of same scan operator
   56: optional bool enable_shared_scan
+
+  57: optional string stream_load_label
```
What's the diffenence of `import_label` and `stream_load_label`?

trueeyu replied:
why not use the `load_job_id` ?

trueeyu replied:
Better not add unused item to thrift which is only temp used on be.

xiaoyong-z commented:
@mergify rebase

meegoo commented on be/src/runtime/stream_load/transaction_mgr.cpp:
```diff
@@ -362,6 +362,25 @@ Status TransactionMgr::_commit_transaction(StreamLoadContext* ctx, bool prepare)
+            if (ctx->unref()) {
+                delete ctx;
+            }
+        });
```
should acquire lock

xiaoyong-z committed:
[BugFix] transaction stream load cancel bug

After we begin a transaction stream load and load, if we don't abort or commit for a long time,
transaction stream load will timeout, however, BE itself cannot cancel stream load.
This is because timeout_second hasn't been set if we don't explicitly specify timeout in stream load.

Signed-off-by: xyz <a997647204@gmail.com>

wanpengfei-git commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_admit_test

wanpengfei-git commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_fe_unittest

xiaoyong-z commented:
run starrocks_be_unittest

xiaoyong-z commented:
run starrocks_admit_test

xiaoyong-z commented:
run starrocks_clang-format

wanpengfei-git commented:
@Mergifyio backport branch-2.5

wanpengfei-git commented:
@Mergifyio backport branch-2.4
